### PR TITLE
Seq refactor

### PR DESCRIPTION
**Pull request by jedwards4b**
Refactor the SEQ test so that it works regardless of initial PE layout.  

Test suite: cime_developer 
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes: 

User interface changes?: 

Code review:

### CODE REVIEW

File: scripts/Testing/Testcases/config_tests.xml [hunk 1]
```diff
@@ -276,6 +276,7 @@
   <test NAME="SEQ">
     <DESC>sequencing bfb test (10 day seq,conc tests)</DESC>
     <INFO_DBUG>1</INFO_DBUG>
+    <BFBFLAG>TRUE</BFBFLAG>
```
Comment (jgfouca) [inline scripts/Testing/Testcases/config_tests.xml]: Who uses this flag?

File: utils/python/CIME/SystemTests/seq.py [hunk 2]
```diff
@@ -14,7 +14,59 @@ def __init__(self, caseroot, case):
         """
         initialize an object interface to file env_test.xml in the case directory
         """
-        SystemTestsCommon.__init__(self, caseroot, case)
+        SystemTestsCommon.__init__(self, caseroot, case, expected=["TEST"])
```
Comment (jgfouca) [inline utils/python/CIME/SystemTests/seq.py]: What does this new expected argument do?

File: utils/python/CIME/SystemTests/seq.py [hunk 3]
```diff
@@ -26,26 +78,27 @@ def run(self):
...
-            for comp in ['ATM','CPL','OCN','WAV','GLC','ICE','ROF','LND', 'ESP']:
-                any_changes |= self._case.get_value("ROOTPE_%s" % comp) != 0
-                self._case.set_value("ROOTPE_%s" % comp, 0)
-
-            expect(any_changes, "All ROOTPEs were already zero, we aren't testing anything")
+        shutil.copy("%s/%s.exe.SEQ1"%(exeroot,cime_model),
+                    "%s/%s.exe"%(exeroot,cime_model))
+        shutil.copy(os.path.join("LockedFiles", "env_mach_pes.SEQ1.xml"), "env_mach_pes.xml")
+        shutil.copy("env_mach_pes.xml", os.path.join("LockedFiles", "env_mach_pes.xml"))
+        success = SystemTestsCommon._run(self)
```
Comment (jgfouca) [inline utils/python/CIME/SystemTests/seq.py]: If success is false, the test should probably halt immediately. That's how I did ERS.

File: utils/python/update_acme_tests.py [hunk 4]
```diff
@@ -64,8 +64,8 @@
 
     "cime_developer" : (None,
                             ("NCK_Ld3.f45_g37_rx1.A",
-                             "ERI_D.f45_g37_rx1.A",
-                             "SEQ_Ld3.f09_g16.X",
+                             "ERI_D.f45_g37.X",
```
Comment (jgfouca) [inline utils/python/update_acme_tests.py]: Why did these need to change?

**Comment by jgfouca** [comment]
Review complete. One code change requested and some questions.

File: scripts/Testing/Testcases/config_tests.xml (hunk 1 above)
Comment (jedwards4b) [inline scripts/Testing/Testcases/config_tests.xml]: The coupler - if the number of coupler tasks change between tests this needs to be set to true to get bfb results.

File: utils/python/CIME/SystemTests/seq.py (hunk 2 above)
Comment (jedwards4b) [inline utils/python/CIME/SystemTests/seq.py]: See my comments in the slack general.   It causes a comparison of env_run.xml with the original to reset this field to the original value.

File: utils/python/CIME/SystemTests/seq.py (hunk 3 above)
Comment (jedwards4b) [inline utils/python/CIME/SystemTests/seq.py]: I'll fix that.

File: utils/python/update_acme_tests.py (hunk 4 above)
Comment (jedwards4b) [inline utils/python/update_acme_tests.py]: The SEQ test makes more sense as an A case, but I wanted to make sure we were still testing an X case.  Also no need to run this test for 3 days, 9 steps works fine.

**Pull request merged**